lib/reek/smells/smell_detector.rb

```diff
@@ -118,6 +113,18 @@ def value(key, ctx, fall_back)
       def config_for(ctx)
         ctx.config_for(self.class)
       end
+
+      def create_smell_warning(options = {})
```

Comment:
Can we drop the "create_" terminology and just use "smell_warning"? Feels java-esque :-(